### ISSUE TITLE
Add Conditional Formatting to Vuu Grids

### ISSUE THREAD

**Issue by chrisjstevo** [venuu-io/vuu#223]

**Comment by heswell** [comment]
I don't think we should be sending explicit styling instructions from the server to the client. When the user picks whatever styling effects they want to be associated with the filter criteria, we should create a custom css classname , store that classname in the layout json for this layout, together with the associated css styles  and let the server know that this is the className  to be applied to this row/column when they meet this set of filter criteria . The client would make sure sure when a layout is loaded, that custom classnames are injected into the loaded stylesheet, all ready to style any cells/rows with the associated classname.
The server just needs to send the associated classname(s) with rows/columns that meet the filter criteria.

One edge case I can imagine - what if a row meets multiple filter criteria that have styles associated with them ? Onus is going to have to be on user to make sure they don't look terrible. 

WOuld be really nice to have a preview capability in the UI that user uses to set these, so they can anticipate this issue.

**Comment by chrisjstevo** [comment]
I agree, so the UI would just send the class and the server would evaluate the criteria and return the class or not..?

Are you happy with how it would return the class?

I would say column trumps row in terms of precedence.

**Comment by heswell** [comment]
agree column before row.

If I'm going to save down the styling effects with the layout, we're going to have to start persisting layouts.
WOuld probably also be nice, for this and other things like named filters, to have a mechanism to save settings that might span multiple layouts.

**Comment by chrisjstevo** [comment]
I'll write up a file system storage version, will take 2 mins.

When do you feel you would store something outside of a layout? My feeling is you want everything in there in case you share it (filters and all) with someone else.
